### PR TITLE
[FLINK-7707] [flip6] Add TaskCheckpointStatisticDetailsHandler for new REST endpoint

### PR DESCRIPTION
**Pull request by tillrohrmann**
## What is the purpose of the change

Adds the `TaskCheckpointStatisticDetailsHandler` which is the equivalent of the `CheckpointStatsDetailsSubtaskHandler` of the old `WebRuntimeMonitor`.

## Verifying this change

This change added tests and can be verified as follows:

- `TaskCheckpointStatisticsTest` and `TaskCheckpointStatisticsWithSubtaskDetailsTest` to test the (un)marshalling of these classes

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/job/checkpoints/TaskCheckpointStatisticDetailsHandler.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+				taskStatistics.getLatestAckTimestamp(),
+				taskStatistics.getStateSize(),
+				taskStatistics.getEndToEndDuration(checkpointStats.getTriggerTimestamp()),
+				taskStatistics.getAlignmentBuffered(),
+				taskStatistics.getNumberOfSubtasks(),
+				taskStatistics.getNumberOfAcknowledgedSubtasks(),
+				summary,
+				subtaskCheckpointStatistics);
+		} else {
+			throw new RestHandlerException("There is no checkpoint statistics for task " + jobVertexId + '.', HttpResponseStatus.NOT_FOUND);
```
Comment (zentol) [inline flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/job/checkpoints/TaskCheckpointStatisticDetailsHandler.java]: "There are"?

Comment (tillrohrmann) [reply flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/job/checkpoints/TaskCheckpointStatisticDetailsHandler.java]: I think statistics should be used with a singular verb, but I'm not entirely sure, either.

**Comment by tillrohrmann** [comment]
Thanks for the review @zentol. I've corrected the compilation error. I'll merge this PR once Travis gives green light and the preceding PR has been merged.

**Comment by tillrohrmann** [comment]
Rebased onto the latest master.

**Pull request closed without merging**